commit 712ad57d09a2a96dc1e21cd245fbf1a6372867ba
Author: John Spray <jspray@redhat.com>
Date:   Thu Nov 16 16:30:39 2017 -0500

    mgr: evaluate `can_run` method on modules
    
    ...and transmit the result to the monitor in
    our beacon.
    
    Fixes: http://tracker.ceph.com/issues/21502
    Signed-off-by: John Spray <john.spray@redhat.com>

diff --git a/src/mgr/PyModuleRegistry.h b/src/mgr/PyModuleRegistry.h
index b821cb5c8ac..3eaa9f64092 100644
--- a/src/mgr/PyModuleRegistry.h
+++ b/src/mgr/PyModuleRegistry.h
@@ -59,6 +59,14 @@ public:
 
   void list_modules(std::set<std::string> *modules);
 
+  void get_modules(std::list<PyModuleRef> *modules_out)
+  {
+    Mutex::Locker l(lock);
+    for (const auto &i : modules) {
+      modules_out->push_back(i.second);
+    }
+  }
+
   PyModuleRegistry(LogChannelRef clog_)
     : clog(clog_)
   {}
